src/main/java/com/google/devtools/build/lib/bazel/repository/GitCloner.java

```diff
@@ -115,6 +116,15 @@ public static SkyValue clone(Rule rule, Path outputDirectory, EventHandler event
         mapper.get("init_submodules", Type.BOOLEAN),
         outputDirectory);
 
+	// Setup proxy if remote is http or https
+    if (descriptor.remote != null && descriptor.remote.startsWith("http")) {
+      try {
+        ProxyHelper.createProxyIfNeeded(new URL(descriptor.remote));
```

Comment:
Do not create a URL.  The URL constructor does a DNS resolution, which is very slow.